src/sources/mongodb_metrics/mod.rs

```diff
@@ -0,0 +1,1065 @@
+                tags!(self.tags, "type" => "uncompressed"),
+            ));
+            metrics.push(self.create_metric(
+                "mongod_wiredtiger_log_records_total",
+                counter!(stat.log.records_processed_log_scan),
+                tags!(self.tags),
+            ));
+
+            // mongod_wiredtiger_session_open_sessions
+            metrics.push(self.create_metric(
+                "mongod_wiredtiger_session_open_sessions",
+                gauge!(stat.session.sessions),
+                tags!(self.tags),
+            ));
+
+            // mongod_wiredtiger_transactions_*
+            metrics.push(self.create_metric(
+                "mongod_wiredtiger_transactions_checkpoint_seconds",
+                gauge!(stat.transaction.checkpoint_min_ms / 1000),
+                tags!(self.tags, "type" => "min"),
+            ));
+            metrics.push(self.create_metric(
+                "mongod_wiredtiger_transactions_checkpoint_seconds",
+                gauge!(stat.transaction.checkpoint_max_ms / 1000),
+                tags!(self.tags, "type" => "max"),
+            ));
+            metrics.push(self.create_metric(
+                "mongod_wiredtiger_transactions_checkpoint_seconds_total",
+                counter!(stat.transaction.checkpoint_total_ms / 1000),
+                tags!(self.tags),
+            ));
+            metrics.push(self.create_metric(
+                "mongod_wiredtiger_transactions_running_checkpoints",
+                gauge!(stat.transaction.checkpoints_running),
+                tags!(self.tags),
+            ));
+            metrics.push(self.create_metric(
+                "mongod_wiredtiger_transactions_total",
+                counter!(stat.transaction.begins),
+                tags!(self.tags, "type" => "begins"),
+            ));
+            metrics.push(self.create_metric(
+                "mongod_wiredtiger_transactions_total",
+                counter!(stat.transaction.checkpoints),
+                tags!(self.tags, "type" => "checkpoints"),
+            ));
+            metrics.push(self.create_metric(
+                "mongod_wiredtiger_transactions_total",
+                counter!(stat.transaction.committed),
+                tags!(self.tags, "type" => "committed"),
+            ));
+            metrics.push(self.create_metric(
+                "mongod_wiredtiger_transactions_total",
+                counter!(stat.transaction.rolled_back),
+                tags!(self.tags, "type" => "rolledback"),
+            ));
+        }
+
+        // network_*
+        metrics.push(self.create_metric(
+            "network_bytes_total",
+            counter!(status.network.bytes_in),
+            tags!(self.tags, "state" => "bytes_in"),
+        ));
+        metrics.push(self.create_metric(
+            "network_bytes_total",
+            counter!(status.network.bytes_out),
+            tags!(self.tags, "state" => "bytes_out"),
+        ));
+        metrics.push(self.create_metric(
```

Comment:
Well, this is a list.

Reply:
Ana, sorry, I'm not sure that I understand what you mean :slightly_frowning_face: do you propose to use LinkedList instead Vec?

Reply:
I'm sorry, I was more in awe at the length!